core/src/main/java/de/danoeh/antennapod/core/feed/LocalFeedUpdater.java

```diff
@@ -0,0 +1,74 @@
+package de.danoeh.antennapod.core.feed;
+
+import android.content.Context;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import de.danoeh.antennapod.core.storage.DBTasks;
+
+public class LocalFeedUpdater {
+
+    private static long getFileDuration(File f) {
+        MediaMetadataRetriever mediaMetadataRetriever = new MediaMetadataRetriever();
+        mediaMetadataRetriever.setDataSource(f.getAbsolutePath());
+        String durationStr = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION);
+        return Long.parseLong(durationStr);
+    }
+
+    /** Starts the import process. */
+    public static void startImport(Uri uri, Context context) {
+        File f = new File(uri.getPath());
+        if (!f.isDirectory()) {
+            throw new RuntimeException("invalid path");
+        } else {
+            startImportDirectory(uri, context);
+        }
+    }
+
+    private static void startImportDirectory(Uri uri, Context context) {
+        File f = new File(uri.getPath());
+        String dirUrl = uri.toString();
+        Feed dirFeed = new Feed(dirUrl, null, "Local directory (" + dirUrl + ")");
+        List<FeedItem> items = new ArrayList<>();
+
+        //find relevant files and create items for them
+        File[] itemFiles = f.listFiles(
+                new FilenameFilter() {
+                    @Override
+                    public boolean accept(File file, String s) {
+                        return s.endsWith(".mp3");
```

Comment:
I suggest to filter by mime type ([source](https://stackoverflow.com/a/8591230/4193263)):

```
    public static String getMimeType(String path) {
        String extention = path.substring(path.lastIndexOf("."));
        String mimeTypeMap = MimeTypeMap.getFileExtensionFromUrl(extention);
        return MimeTypeMap.getSingleton().getMimeTypeFromExtension(mimeTypeMap);
    }
```

AntennaPod can likely handle all `audio/...` and `video/...` types

Reply:
Nice!